### ISSUE TITLE
Method importFromArray from \Magento\Eav\Model\Entity\Collection\AbstractCollection doesn't return a working collection

### ISSUE THREAD

**Issue by slackerzz** [magento/magento2#21868]
### Preconditions (*)

1. Create a collection, for example using `\Magento\Catalog\Model\ResourceModel\Product\CollectionFactory`:

```php
$skus = ['sku1', 'sku2'];
$collection = $this->collectionFactory->create();
$collection->addAttributeToFilter('sku', ['in' => $skus]);
// assuming that products `sku1' and `sku2` exist 
```

2. Export collection data as a 2D array with `exportToArray()`
```php
$collectionData = $collection->exportToArray();
```

### Steps to reproduce (*)

1. Now create a new collection and try to use `importFromArray()` to populate it
```php
$newCollection = $this->collectionFactory->create();
$newCollection->importFromArray($collectionData)
```

### Expected result (*)

1. `$newCollection->getItems()` should return an array with the 2 products

### Actual result (*)

1. `$newCollection->getItems()` throw an exception
```
1 exception(s):
Exception #0 (Exception): Item (Magento\Catalog\Model\Product\Interceptor) with the same ID "7897" already exists.
```

**Comment by magento-engcom-team** [comment]
Hi @slackerzz. Thank you for your report. 
To help us process this issue please make sure that you provided the following information:
- [ ] Summary of the issue
- [ ] Information on your environment
- [x] Steps to reproduce
- [x] Expected and actual results

Please make sure that the issue is reproducible on the vanilla Magento instance following **Steps to reproduce**. To deploy vanilla Magento instance on our environment, please, add a comment to the issue: 

`@magento-engcom-team give me 2.3-develop instance` - upcoming 2.3.x release

For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation.

@slackerzz do you confirm that you was able to reproduce the issue on vanilla Magento instance following steps to reproduce?
- [x] yes
- [ ] no

**Comment by magento-engcom-team** [comment]
Hi @engcom-backlog-nazar. Thank you for working on this issue. 
In order to make sure that issue has enough information and ready for development, please read and check the following instruction: :point_down:
- [x] 1. Verify that issue has all the required information. (**Preconditions**, **Steps to reproduce**, **Expected result**, **Actual result**).<details><summary>Details</summary>If the issue has a valid description, the label `Issue: Format is valid` will be added to the issue automatically. Please, edit issue description if needed, until label `Issue: Format is valid` appears.</details>

- [x] 2. Verify that issue has a meaningful description and provides enough information to reproduce the issue. If the report is valid, add `Issue: Clear Description` label to the issue by yourself.

- [x] 3. Add `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.
- [x] 4. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [x] 5. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

-  _Next steps are available in case you are a member of **Community Maintainers**._

- [x] 6. Add label `Issue: Confirmed` once verification is complete. 

- [x] 7. Make sure that automatic system confirms that report has been added to the backlog.

**Comment by engcom-backlog-nazar** [comment]
HI @slackerzz can you please explain more detail, i'm not able to reproduce.
![DeepinScreenshot_select-area_20190321125402](https://user-images.githubusercontent.com/40756439/54747785-818c8580-4bd8-11e9-8f8f-b60ba69ce730.png)
![DeepinScreenshot_select-area_20190321125411](https://user-images.githubusercontent.com/40756439/54747787-82251c00-4bd8-11e9-9270-a02cc496ca79.png)

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @engcom-backlog-nazar 
 Thank you for verifying the issue. Based on the provided information internal tickets `MAGETWO-98828`, `MAGETWO-98829` were created

**Issue Available**: @engcom-backlog-nazar, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by hmfilar** [comment]
#snowdog-oss

**Comment by magento-engcom-team** [comment]
@HMF-HFilar thank you for joining. Please accept team invitation [here](https://github.com/orgs/magento/invitation) and self-assign the issue.

**Comment by magento-engcom-team** [comment]
Hi @HMF-HFilar. Thank you for working on this issue. 
Looks like this issue is already verified and confirmed. But if you want to validate it one more time, please, go though the following instruction:
- [ ] 1. Add/Edit `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.

- [ ] 2. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [ ] 3. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

- [ ] 4. If the issue is not relevant or is not reproducible any more, feel free to close it.

**Comment by magento-engcom-team** [comment]
Hi @slackerzz. Thank you for your report. 
The issue has been fixed in magento/magento2#21869 by @slackerzz in 2.3-develop branch
Related commit(s):
 - [2f49a5c430d1fc805ef0471b4d604d58b6ba0234](https://github.com/magento/magento2/commit/2f49a5c430d1fc805ef0471b4d604d58b6ba0234)
 - [3773c27d3f0d75ee2cf7a421c2c3453f96b9c02c](https://github.com/magento/magento2/commit/3773c27d3f0d75ee2cf7a421c2c3453f96b9c02c)
 - [0de457220574931d82e11be83b4f565f38ab06cb](https://github.com/magento/magento2/commit/0de457220574931d82e11be83b4f565f38ab06cb)

The fix will be available with the upcoming 2.3.2 release.

**Comment by magento-engcom-team** [comment]
Hi @slackerzz. Thank you for your report. 
The issue has been fixed in magento/magento2#22422 by @slackerzz in 2.2-develop branch
Related commit(s):
 - [e1f91d1a87dabac77264f40992007124b8febd4e](https://github.com/magento/magento2/commit/e1f91d1a87dabac77264f40992007124b8febd4e)

The fix will be available with the upcoming 2.2.9 release.
